src/examples/libpmemobj/cpp_map/ctree_map_persistent.hpp

```diff
@@ -203,15 +203,15 @@ class ctree_map_p {
 	{
 		auto pop = nvobj::pool_by_vptr(this);
 		nvobj::transaction::exec_tx(pop, [&] {
-			if (root->inode) {
-				root->inode->clear();
-				nvobj::delete_persistent<node>(root->inode);
-				root->inode = nullptr;
+			if (this->root->inode) {
```

Comment:
I have seen this and it seems to be an issue with how the Microsoft compiler resolves templates inside lambdas. At least I think so. Maybe someone with more experience with MSVC can shed some light on this matter.